File: 2020/old/xgraphics/src/js/homeworld2/goblinmesh.js
import Obj from '../x/graphics/obj.js';

export default class Goblinmesh extends Obj
{
    constructor(name, parent, status)
    {
        super(name, 0, status);

        this.__singlemesh = null;
        this.__parent = parent;
    }

    get singlemesh(){ return this.__singlemesh; }
    get status(){ return this.__status; }
    get name(){ return this.__name; }
    get parent(){ return this.__parent; }

    set singlemesh(o){ this.__singlemesh = o; }

    set status(v)
    {
        this.__status = v;
        this.__singlemesh.status = v;
    }

    render(application, model, level)
    {
        this.__singlemesh.render(application, model, level);

    }

    load(application)
    {
        this.__singlemesh.load(application);
        this.__load = true;
    }


    unload(application)
    {
        this.__singlemesh.unload(application);
        this.__load = false;
    }
}

File: 2020/old/discomplexity/homeworld2/src/Homeworld2/Goblinmesh.cc
#include "Goblinmesh.hh"

namespace Homeworld2
{
    Goblinmesh::Goblinmesh()
    :   x::graphics::object(),
        __singlemesh(),
        __parent()
    {
    }

    Goblinmesh::Goblinmesh(const Goblinmesh & o)
    :   x::graphics::object(o),
        __singlemesh(o.__singlemesh),
        __parent(o.__parent)
    {
    }

    Goblinmesh::Goblinmesh(Goblinmesh && o) noexcept
    :   x::graphics::object(x::move(o)),
        __singlemesh(x::move(o.__singlemesh)),
        __parent(x::move(o.__parent))
    {
    }

    Goblinmesh::~Goblinmesh()
    {
    }

    Goblinmesh & Goblinmesh::operator=(const Goblinmesh & o)
    {
        if(&o != this)
        {
            x::graphics::object::operator=(o);
            __singlemesh = o.__singlemesh;
            __parent = o.__parent;
        }
        return *this;
    }

    Goblinmesh & Goblinmesh::operator=(Goblinmesh && o) noexcept
    {
        if(&o != this)
        {
            x::graphics::object::operator=(x::move(o));
            __singlemesh = x::move(o.__singlemesh);
            __parent = x::move(o.__parent);
        }
        return *this;
    }

    void Goblinmesh::load()
    {
        __singlemesh.load();
    }
    void Goblinmesh::unload()
    {
        __singlemesh.unload();
    }
    void Goblinmesh::render(const x::graphics::matrix & model, x::uint32 level)
    {
        __singlemesh.render(model, level);
    }

    void Goblinmesh::status(x::uint32 v)
    {
        __singlemesh.status(v);
    }
}
